In [1]:
from dgp import *

# 1. Simulate or load data
sample_size = 800
min_degree = 5
max_degree = 7
adj = get_graph(sample_size, min_degree, max_degree, seed=1)

tau = np.array([-1.0, 0.50, -0.50])       # shape (3,)
rho = np.array([[0,0.1,0.2],
                [0.1,0,0.1],
                [0.2,0.1,0]])      # shape (3, 3), with 0s on the diagonal
nu = np.array([0.1,0,0,0.1,0,0,0.1,0,0]).reshape(3,3)       # shape (3, 3)
gamma = np.array([-1.00,0.50,0.10,0.20,0.05,0.25,-0.08,0.30])     # shape (8,)
beta = np.array([-0.30,-0.60,-0.20,-0.20,-0.05,-0.10,-0.01,0.40,0.01,0.20])     # shape (10,)

Y_chain, A_chain, L_chain = sample_network_chain(adj, tau, rho, nu, gamma, beta, R=500,
    burnin_R=20, seed=0)

Y_chain = Y_chain[::5]
A_chain = A_chain[::5]
L_chain = L_chain[::5]

100%|██████████| 520/520 [00:17<00:00, 30.27it/s]


In [9]:
from agcEffect import *

agc_effect(
    adj,
    tau, rho, nu, beta,
    treatment_allocation=0.7,
    R=200,
    burnin_R=50,
    seed=0
)

{'average': 0.21255625000000003,
 'direct_effect': -0.13160625000000006,
 'spillover_effect': -0.26986874999999994,
 'psi_1_gamma': 0.17561874999999996,
 'psi_0_gamma': 0.307225,
 'psi_zero': 0.57709375}

In [3]:
n_cpu = 10

In [11]:
from utils import run_pll
from run_pll import run_dr

args = [{'Y_chain': Y_chain,
         'A_chain': A_chain,
         'L_chain': L_chain,
         'adj': adj,
         'i': i} for i in range(len(Y_chain))]

res_list_dr = run_pll(run_dr, args, processes=n_cpu)
res_list_array_dr = np.array(res_list_dr)
# save results
np.save(f'run/sim_results/sim_dr_{sample_size}_{min_degree}_{max_degree}.npy', res_list_array_dr)

Multiprocessing <function run_dr at 0x10e96a710> in 100 tasks, with 10 processes...
Multiprocessing finished.


In [12]:
from run_pll import column_names

ret_mean_dr = res_list_array_dr.mean(axis=0)
ret_std_dr = res_list_array_dr.std(axis=0)
for i in range(len(column_names)):
    print(f"{column_names[i]}: {ret_mean_dr[i]:.5f} ± {ret_std_dr[i]:.5f}")

average: 0.21446 ± 0.02133
direct_effect: -0.10792 ± 0.04988
spillover_effect: -0.07699 ± 0.81592
psi_0_gamma: 0.28908 ± 0.04498
psi_zero: 0.36607 ± 0.81492
psi_1_gamma: 0.18116 ± 0.02288


In [13]:
from utils import run_pll
from run_pll import run_autognet

args = [{'Y_chain': Y_chain,
         'A_chain': A_chain,
         'L_chain': L_chain,
         'adj': adj,
         'i': i} for i in range(len(Y_chain))]

res_list_ag = run_pll(run_autognet, args, processes=n_cpu)
res_list_array_ag = np.array(res_list_ag)
# save results
np.save(f'run/sim_results/sim_autog_{sample_size}_{min_degree}_{max_degree}.npy', res_list_array_ag)

Multiprocessing <function run_autognet at 0x17f8b8e50> in 100 tasks, with 10 processes...
Multiprocessing finished.


In [14]:
from run_pll import column_names

ret_mean_ag = res_list_array_ag.mean(axis=0)
ret_std_ag = res_list_array_ag.std(axis=0)
for i in range(len(column_names)):
    print(f"{column_names[i]}: {ret_mean_ag[i]:.5f} ± {ret_std_ag[i]:.5f}")

average: 0.21087 ± 0.02507
direct_effect: -0.14345 ± 0.05489
spillover_effect: -0.12993 ± 0.08235
psi_0_gamma: 0.31975 ± 0.05226
psi_zero: 0.44968 ± 0.10073
psi_1_gamma: 0.17630 ± 0.02557


{'average': 0.21255625000000003,
 'direct_effect': -0.13160625000000006,
 'spillover_effect': -0.26986874999999994,
 'psi_1_gamma': 0.17561874999999996,
 'psi_0_gamma': 0.307225,
 'psi_zero': 0.57709375}


average: 0.21446 ± 0.02133
direct_effect: -0.10792 ± 0.04988
spillover_effect: -0.07699 ± 0.81592
psi_0_gamma: 0.28908 ± 0.04498
psi_zero: 0.36607 ± 0.81492
psi_1_gamma: 0.18116 ± 0.02288

In [4]:
from utils import run_pll
from run_pll import run_ocnet

args = [{'Y_chain': Y_chain,
         'A_chain': A_chain,
         'L_chain': L_chain,
         'adj': adj,
         'i': i} for i in range(len(Y_chain))]

res_list_ag = run_pll(run_ocnet, args, processes=n_cpu)
res_list_array_ag = np.array(res_list_ag)
# save results
np.save(f'run/sim_results/sim_ocnet_{sample_size}_{min_degree}_{max_degree}.npy', res_list_array_ag)

Multiprocessing <function run_ocnet at 0x13eee1120> in 100 tasks, with 10 processes...
Multiprocessing finished.


In [5]:
from run_pll import column_names

ret_mean_ag = res_list_array_ag.mean(axis=0)
ret_std_ag = res_list_array_ag.std(axis=0)
for i in range(len(column_names)):
    print(f"{column_names[i]}: {ret_mean_ag[i]:.5f} ± {ret_std_ag[i]:.5f}")

average: 0.21162 ± 0.01666
direct_effect: -0.09426 ± 0.03208
spillover_effect: -0.18070 ± 0.06849
psi_0_gamma: 0.27758 ± 0.02974
psi_zero: 0.45829 ± 0.07281
psi_1_gamma: 0.18333 ± 0.01806
